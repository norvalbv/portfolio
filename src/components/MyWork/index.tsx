import React, { ReactElement } from 'react';
import CardWrapper from 'components/CardWrapper';
import CodeLink from 'components/CodeLink';
import { parisnDescription, parisnSkills, weatherAppSkills } from 'constants';
import StyledLink from 'components/StyledLink';
import WeatherApp from '../../../public/WeatherAppHomepage.png';
import ParisnHomepage from '../../../public/ParisnHomepage.png';

const MyWork = (): ReactElement => {
  return (
    <CardWrapper title="My Work">
      <div className="relative mb-10 rounded-t-xl bg-gray-700/75 px-6 py-10 text-center before:absolute before:bottom-[-0.25rem] before:left-0 before:h-1 before:w-full before:bg-gradient-to-r before:from-blue-500 before:to-green-500">
        <h3 className="mb-6 text-xl uppercase underline">
          &apos;PARISN&apos; - An e-commerce application with a twist
        </h3>
        <CodeLink link="https://github.com/norvalbv/parisn" className="absolute right-12 top-12" />
        <p>{parisnDescription}</p>
        <div>
          <img
            src={ParisnHomepage}
            alt="Parisn App Homepage - An ecommerce site."
            className="mx-auto my-6 max-h-[35rem]"
          />
          <p>Built with: {parisnSkills}</p>
        </div>
      </div>
      <div className="relative my-10 rounded-t-xl bg-gray-700/75 px-6 py-10 text-center before:absolute before:bottom-[-0.25rem] before:left-0 before:h-1 before:w-full before:bg-gradient-to-r before:from-blue-500 before:to-green-500">
        <h3 className="mb-6 text-xl uppercase underline">Weather App</h3>
        <CodeLink
          link="https://github.com/norvalbv/cool-weather-app"
          className="absolute right-12 top-12"
        />
        <p>
          A front end weather application built with the&nbsp;
          <StyledLink label="Open Weather Api" link="https://openweathermap.org/api" />
          &nbsp;that uses live data and the ability to search for weather worldwide.
        </p>
        <div>
          <img src={WeatherApp} alt="Weather App Homepage" className="mx-auto my-6 max-h-[35rem]" />
          <p>Built with: {weatherAppSkills}</p>
        </div>
      </div>
    </CardWrapper>
  );
};

export default MyWork;
