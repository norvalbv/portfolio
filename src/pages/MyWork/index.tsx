import React, { ReactElement } from 'react';
import CardWrapper from 'components/CardWrapper';
import CodeLink from 'components/CodeLink';
import { PARISN_DESCRIPTION, PARISN_SKILLS, WEATHER_APP_SKILL } from 'constants/index';
import StyledLink from 'components/StyledLink';
import WeatherApp from '../../../public/WeatherAppHomepage.png';
import ParisnHomepage from '../../../public/ParisnHomepage.png';

const MyWork = (): ReactElement => {
  return (
    <CardWrapper
      title="My Work"
      className="top-[4rem] min-w-[20rem] md:w-10/12"
      titleClassName="md:ml-0 ml-6"
    >
      <div className="relative mb-10 rounded-t-xl bg-dark-dark/20 py-6 text-center before:absolute before:bottom-[-0.25rem] before:left-0 before:h-1 before:w-full before:bg-gradient-to-r before:from-blue-500 before:to-green-500 dark:bg-dark-dark/30 md:px-6">
        <h3 className="mb-6 px-6 text-lg uppercase underline md:p-0 lg:text-xl">
          &apos;PARISN&apos; - An e-commerce application with a twist
        </h3>
        <CodeLink
          link="https://github.com/norvalbv/parisn"
          className="my-6 lg:absolute lg:right-12 lg:top-1"
        />
        <p className="px-6 text-sm md:p-0 lg:text-base">{PARISN_DESCRIPTION}</p>
        <a href="https://github.com/norvalbv/parisn" rel="noopener noreferrer" target="_blank">
          <img
            src={ParisnHomepage}
            alt="Parisn App Homepage - An ecommerce site."
            className="mx-auto my-6 max-h-[40rem] min-h-[14em]"
          />
        </a>
        <p className="px-6 text-sm md:p-0 lg:text-base">Built with: {PARISN_SKILLS}</p>
      </div>
      <div className="relative my-10 rounded-t-xl bg-dark-dark/20 py-6 text-center before:absolute before:bottom-[-0.25rem] before:left-0 before:h-1 before:w-full before:bg-gradient-to-r before:from-blue-500 before:to-green-500 dark:bg-dark-dark/30 md:px-6">
        <h3 className="mb-6 px-6 text-lg uppercase underline md:p-0 lg:text-xl">Weather App</h3>
        <CodeLink
          link="https://github.com/norvalbv/cool-weather-app"
          className="my-6 lg:absolute lg:right-12 lg:top-1"
        />
        <p className="px-6 text-sm md:p-0 lg:text-base">
          A front end weather application built with the&nbsp;
          <StyledLink label="Open Weather Api" link="https://openweathermap.org/api" />
          &nbsp;that uses live data and the ability to search for weather worldwide.
        </p>
        <a
          href="https://github.com/norvalbv/cool-weather-app"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            src={WeatherApp}
            alt="Weather App Homepage"
            className="mx-auto my-6 max-h-[35rem] min-h-[14em]"
          />
        </a>
        <p className="px-6 text-sm md:p-0 lg:text-base">Built with: {WEATHER_APP_SKILL}</p>
      </div>
    </CardWrapper>
  );
};

export default MyWork;
