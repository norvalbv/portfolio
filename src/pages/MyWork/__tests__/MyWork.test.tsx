import React from 'react';
import renderer from 'react-test-renderer';

import MyWork from '..';

describe('<MyWork />', () => {
  test('should render correctly', () => {
    const tree = renderer.create(<MyWork />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
