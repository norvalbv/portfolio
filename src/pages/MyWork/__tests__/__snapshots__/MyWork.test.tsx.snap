// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyWork /> should render correctly 1`] = `
<div
  className="relative mx-auto flex-1 mt-6 w-10/12 min-w-[20rem] md:mt-10"
>
  <div>
    <header>
      <section
        className="flex w-full flex-col gap-1"
      >
        <h1
          className="md:ml-0 ml-6 mb-[4.5rem] w-max border-b pb-2 text-5xl font-semibold md:text-6xl"
        >
          H
          e
          l
          l
          o
           
          W
          o
          r
          l
          d
          !
        </h1>
      </section>
    </header>
    <div
      className="relative mb-10 rounded-t-xl border-light-text bg-black/20 py-6 text-center before:absolute before:bottom-[-0.25rem] before:left-0 before:h-1 before:w-full before:bg-gradient-to-r before:from-blue-500 before:to-green-500 dark:border-dark-text dark:bg-dark-dark/30 md:px-6"
    >
      <h3
        className="px-6 text-lg uppercase underline md:p-0 lg:text-xl"
      >
        PARISN
      </h3>
      <h2
        className="mb-6 mt-2 text-sm uppercase underline lg:text-lg"
      >
        An e-commerce application with a twist
      </h2>
      <div
        className="group my-6 lg:absolute lg:right-12 lg:top-1"
      >
        <a
          className="items-center justify-center gap-2 lg:flex"
          href="https://github.com/norvalbv/parisn"
          rel="noopener noreferrer"
          target="_blank"
        >
          <p
            className="text-xs opacity-100 transition-opacity duration-300 lg:opacity-0 lg:group-hover:opacity-100"
          >
            View Code
          </p>
          <svg
            className="hidden cursor-pointer lg:block"
            fill="none"
            height={20}
            stroke="currentColor"
            strokeWidth={1.5}
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.25 6.75L22.5 12l-5.25 5.25m-10.5 0L1.5 12l5.25-5.25m7.5-3l-4.5 16.5"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </a>
      </div>
      <p
        className="px-6 text-sm md:p-0 lg:text-base"
      >
        A work-in-progress e-commerce full stack webapp with unique add-ons including real-time data and chat using web sockets and automatic pricing that decreases over time using mathematical algorithms giving a better user experience.
      </p>
      <a
        href="https://github.com/norvalbv/parisn"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Parisn App Homepage - An ecommerce site."
          className="mx-auto my-6 max-h-[40rem] min-h-[14em]"
          src="test-file-stub"
        />
      </a>
      <p
        className="px-6 text-sm md:p-0 lg:text-base"
      >
        Built with: 
        React, SWR, JavaScript, TypeScript, TailWindCSS, Vite, Jest, Cypress, Node, AWS Serverless, Azure Devops, CI/CD, Stripe, WebSockets, Microservices.
      </p>
    </div>
    <div
      className="relative mb-10 rounded-t-xl border-light-text bg-black/20 py-6 text-center before:absolute before:bottom-[-0.25rem] before:left-0 before:h-1 before:w-full before:bg-gradient-to-r before:from-blue-500 before:to-green-500 dark:border-dark-text dark:bg-dark-dark/30 md:px-6"
    >
      <h3
        className="px-6 text-lg uppercase underline md:p-0 lg:text-xl"
      >
        Weather App
      </h3>
      <div
        className="group my-6 lg:absolute lg:right-12 lg:top-1"
      >
        <a
          className="items-center justify-center gap-2 lg:flex"
          href="https://github.com/norvalbv/cool-weather-app"
          rel="noopener noreferrer"
          target="_blank"
        >
          <p
            className="text-xs opacity-100 transition-opacity duration-300 lg:opacity-0 lg:group-hover:opacity-100"
          >
            View Code
          </p>
          <svg
            className="hidden cursor-pointer lg:block"
            fill="none"
            height={20}
            stroke="currentColor"
            strokeWidth={1.5}
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.25 6.75L22.5 12l-5.25 5.25m-10.5 0L1.5 12l5.25-5.25m7.5-3l-4.5 16.5"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </a>
      </div>
      <p
        className="px-6 text-sm md:p-0 lg:text-base"
      >
        A front end weather application built with the 
        <a
          className="text-accent-secondary hover:underline"
          href="https://openweathermap.org/api"
          rel="noreferrer noopener"
          target="_blank"
        >
          Open Weather Api
        </a>
         that uses live data and the ability to search for weather worldwide.
      </p>
      <a
        href="https://github.com/norvalbv/cool-weather-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Weather App Homepage."
          className="mx-auto my-6 max-h-[40rem] min-h-[14em]"
          src="test-file-stub"
        />
      </a>
      <p
        className="px-6 text-sm md:p-0 lg:text-base"
      >
        Built with: 
        React, SWR, JavaScript, TypeScript, TailWindCSS, Vite, Jest
      </p>
    </div>
  </div>
</div>
`;
